src/UI/templates/js/Dropzone/File/dropzone.js

```diff
@@ -0,0 +1,267 @@
+            "darkenedDropzoneHighlight": "darkened-highlight",
+            "defaultDropzoneHighlight": "default-highlight",
+            "dropzoneDragHover": "drag-hover"
+        };
+
+        /**
+         * Contains all css selectors used for dropzone manipulation.
+         * Selectors MUST be declared like in css.
+         * e.g.
+         *  .this-is-a-class
+         *  #this-is-a-id
+         */
+        var SELECTOR = {
+            "darkenedBackground": "#il-dropzone-darkened",
+            "dropzones": ".il-dropzone"
+        };
+
+        /**
+         * Contains all supported dropzone types.
+         * The type MUST be equal to the full qualified class name used in php.
+         * NOTE backslashes needs to be removed.
+         * e.g. ILIAS\UI\Component\Dropzone\Standard -> Standard
+         */
+        var DROPZONE = {
+            "standard": "Standard",
+            "wrapper": "Wrapper"
+        };
+
+        var _darkenedBackground = false;
+
+        /**
+         * Initializes a dropzone depending on the passed in type with the passed in options.
+         *
+         * @param {string} type the type of the dropzone
+         *                      MUST be the full qualified class name.
+         * @param {Object} options possible settings for this dropzone
+         */
+        var initializeDropzone = function (type, options) {
+
+            // disable default behavior of browsers for file drops
+            $(document).on("dragenter dragstart dragend dragleave dragover drag drop", function (e) {
+                e.preventDefault();
+            });
+
+            var settings = $.extend({
+                // default settings
+                registeredSignals: [],
+                darkenedBackground: false,
+                uploadUrl: '',
+                uploadButton: null
+            }, options);
+
+            if (settings.id === undefined) {
+                throw new Error("Missing dropzone id in parameter options: options.id not found");
+            }
+
+            switch (type) {
+                case DROPZONE.standard:
+                    _initStandardDropzone(settings);
+                    break;
+                case DROPZONE.wrapper:
+                    _configureDarkenedBackground(true);
+                    _initWrapperDropzone(settings);
+                    break;
+                default:
+                    throw new Error("Unsupported dropzone type found: " + type);
+            }
+
+        };
+
+        /**
+         * Adds a html div to enable the darkened background, if the passed in argument is true.
+         * Sets the state of the darkened background availability to the value of the passed in argument.
+         *
+         * @param {boolean} darkenedBackground true, if the darkened background should be available
+         *
+         * @private
+         */
+        var _configureDarkenedBackground = function (darkenedBackground) {
```

Comment:
Where does the underline come from? (just curious). If you believe this should be a general pattern of handling closures hidden in some scope (private) then please propose that in a PR to the JS section of the frameworks readme.